src/main/rx/eleres.c

```diff
@@ -0,0 +1,854 @@
+/*
+ This file is part of eLeReS (by Michal Maciakowski - Cyberdrones.com).
+ Adapted for MultiWii by Mis (Romuald Bialy)
+ Ported to STM by Marbalon (Marcin Baliniak)
+ */
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <string.h>
+#include "platform.h"
+
+#include "build/build_config.h"
+#include "build/debug.h"
+#include "build/version.h"
+
+#include "common/maths.h"
+#include "common/axis.h"
+#include "common/color.h"
+#include "common/utils.h"
+
+#include "drivers/gpio.h"
+#include "drivers/sensor.h"
+#include "drivers/system.h"
+#include "drivers/serial.h"
+#include "drivers/compass/compass.h"
+#include "drivers/timer.h"
+#include "drivers/time.h"
+#include "drivers/pwm_rx.h"
+#include "drivers/accgyro/accgyro.h"
+#include "drivers/light_led.h"
+#include "drivers/sound_beeper.h"
+#include "drivers/io.h"
+#include "drivers/rx_spi.h"
+
+#include "fc/rc_controls.h"
+#include "io/gimbal.h"
+#include "io/gps.h"
+#include "io/ledstrip.h"
+#include "io/serial.h"
+#include "fc/cli.h"
+#include "io/statusindicator.h"
+#include "io/osd.h"
+
+#include "rx/rx.h"
+#include "rx/msp.h"
+#include "rx/rx_spi.h"
+#include "rx/eleres.h"
+
+#include "telemetry/telemetry.h"
+#include "flight/mixer.h"
+#include "flight/failsafe.h"
+#include "flight/imu.h"
+#include "flight/pid.h"
+
+#include "navigation/navigation.h"
+
+#include "sensors/battery.h"
+#include "sensors/sensors.h"
+#include "sensors/barometer.h"
+
+#include "io/beeper.h"
+#include "drivers/bus_spi.h"
+#include "fc/runtime_config.h"
+
+#include "config/config_master.h"
+#include "config/parameter_group.h"
+#include "config/parameter_group_ids.h"
+#include "scheduler/scheduler.h"
+
+#ifdef USE_RX_ELERES
+
+
+PG_REGISTER_WITH_RESET_TEMPLATE(eleresConfig_t, eleresConfig, PG_ELERES_CONFIG, 0);
+
+PG_RESET_TEMPLATE(eleresConfig_t, eleresConfig,
+                  .eleres_freq = 435,
+                  .eleres_telemetry_en = 0,
+                  .eleres_telemetry_power = 7,
+                  .eleres_loc_en = 0,
+                  .eleres_loc_power = 7,
+                  .eleres_loc_delay = 240
+                 );
+
+/*****************    eLeReS compatibile reciver on RFM22B module  ********************/
+rx_spi_received_e Rfm_IRQ(void);
+static uint8_t red_led=0;
+#define RED_LED_ON {if (!red_led) {rfm_spi_write(0x0e, 0x04);red_led=1;}}
+#define RED_LED_OFF {if (red_led) {rfm_spi_write(0x0e, 0x00);red_led=0;}}
+
+#define RC_CHANS 12
+
+static uint8_t hopping_channel = 1;
+
+//int16_t eleresData[RC_CHANS] = { 1500, 1500, 1500, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000, 1000};
+
+void RF22B_init_parameter(void);
+void to_rx_mode(void);
+void ChannelHopping(uint8_t hops);
+void telemetry_RX(void);
+void rx_reset(void);
+void to_tx_mode(uint8_t bytes_to_send);
+void to_ready_mode(void);
+void frequency_configurator(uint32_t frequency);
```

Comment:
We prefer to avoid forward declaration of functions, instead change order of function definitions so that forward declarations are not necessary.